### ISSUE TITLE
Wizard: "Continue" on step 3 can be clicked and nothing will happen

### ISSUE THREAD

**Issue by joaoantoniocardoso** [bluerobotics/BlueOS#2413]
On step 3 of the Wizard, the user can click the CONTINUE button without interacting with the combobox first, which is confusing.

```Vue
              <v-btn
                color="primary"
                @click="validateParams() && setupConfiguration()"
              >
                Continue
              </v-btn>
```

- This CONTINUE button should be disabled if the validateParams are not validated.
- The validation should be lazy (happens the first time only after the user interacts with the combobox)

**Comment by joaoantoniocardoso** [comment]
Still valid on master

**Comment by patrickelectric** [comment]
Unable to replicate, closing for now.

<img width="1404" height="1008" alt="Image" src="https://github.com/user-attachments/assets/ce9ab603-cac2-45a3-8b56-0584d2195fc1" />

**Comment by joaoantoniocardoso** [comment]
This issue is about the bad UX in the wizard, not about a bug:
* the continue button can be clicked indefinitely (not disabled)
* after clicking it, the user is penalized with the validation in red.

This is not a pleasent UX.

**Comment by patrickelectric** [comment]
I'm open for suggestions.

**Comment by joaoantoniocardoso** [comment]
The main pain with this step 3 is that it deceives the user with a blue "continue" button, leading the user to a validation error when they click on it.

There's no single solution here, and it's not something in my domain of knowledge, but I've written below a few points that may help. Note that it is almost inevitable not to expand the discussion to the whole wizard, so maybe we could benefit the whole wizard from this discussion

1. lazy-validate each input
2. disable the scripts dropdown when there are no scripts available, and tell why. The phrasing should be something to give the user confidence that they didn't do anything wrong that caused this. Maybe: "Scripts are optional, there are no scripts available for {vehicle type}."
3. Disable the continue button until the form is valid, not only while it's downloading the scripts/configs, and tell the user why it's disabled and/or what he needs to do. Maybe "Select the default parameters to be loaded accordingly to your vehicle configuration". The options should be named in an easy way and consistent with our website/store. `BlueBoat120` seems weird to me.
4. The user may not know what a script does, and we don't tell what each script does, so we need to provide a link or context somehow.
5. If we need an external guide to be followed with the wizard, we should tell the user, and link each step to a page/section. 

For the boat example: if the navlight script is recommended when the user has a navlight installed, this semantics should be embedded in the wizard flow.

Perhaps @ES-Alexander can give a hand here?

**Comment by ES-Alexander** [comment]
> 2. disable the scripts dropdown when there are no scripts available, and tell why. The phrasing should be something to give the user confidence that they didn't do anything wrong that caused this.

Agreed - this has been raised internally by the SMS product experience team too, though I can't remember if I've reported it before/yet.

I'd suggest something like "Blue Robotics does not have recommended scripts for {firmware type}". That makes it clear it's just a suggestion, and if they bought a vehicle from us and are getting that message it encourages them to ask us about it (e.g. in case there's a new stable firmware release without us updating [the repo](https://github.com/bluerobotics/Blueos-Parameter-Repository/tree/master)).

Another option would be to just hide the selector entirely if it doesn't have any options, though we may want users to know they're not using one of our standard setups, and hiding may require a loading period+indicator for the initial fetch, which is likely not worth the effort to develop and validate/maintain.

> 3. Disable the continue button until the form is valid, not only while it's downloading the scripts/configs, and tell the user why it's disabled and/or what he needs to do. ... The options should be named in an easy way and consistent with our website/store.

This also makes sense. If pressing the button will immediately raise an error then it's a nicer UX to just disable the button until the inputs are valid (assuming that's not problematic to pre-determine in the frontend).

Text descriptor while the button is disabled could be as simple as "Choose your defaults to continue".

> 4. The user may not know what a script does, and we don't tell what each script does, so we need to provide a link or context somehow.

This would be a nice quality of life improvement, though it seems lower priority than the earlier points.

> 5. If we need an external guide to be followed with the wizard, we should tell the user, and link each step to a page/section.

Ideally no external resources should be necessary to refer to as part of a guided setup - otherwise the wizard itself should maybe not exist at all. We could have default collapsed/hidden description sections if necessary, that could be expanded or clicked/hovered over an info icon to view.
